### PR TITLE
fix(tocco-ui): fix filtering of select boxes

### PR DESCRIPTION
**Pull request by PadPalon**
Refs: TOCDEV-5003
Changelog: Fixed searching in select boxes. Removed default searchOptions from Select.
Cherry-pick: Up

### CODE REVIEW

File: packages/tocco-ui/src/Select/Select.js [hunk 1]
```diff
@@ -48,9 +48,12 @@ const Select = ({
 
   const getOptions = () => [...(options || [])]
 
+  const searchFunction = useCallback(searchOptions || (() => { /* _throttle expects a function */ }), [searchOptions])
+  const throttledSearchFunction = useCallback(_throttle(searchFunction, 800, {trailing: true}), [searchFunction])
```
Comment (izzappel) [inline packages/tocco-ui/src/Select/Select.js:52]: I'm not sure if the `_trottle` really works when you have `[searchFunction]` as a deps here.. or.. if you have `[searchOptions]` as a deps on the line above.
I can remember that there was a problem with that and that the _throttle was not working anymore.

Comment (PadPalon) [reply packages/tocco-ui/src/Select/Select.js:52]: Anything I need to test in particulat? Handling of remote fields and select boxes seemed pretty similar to test300 while testing. @yilkur

Comment (izzappel) [reply packages/tocco-ui/src/Select/Select.js:52]: Yes, there should be a delay when typing before the lookup request is triggered. So the lookup request should not get triggered on every keystroke.

Comment (PadPalon) [reply packages/tocco-ui/src/Select/Select.js]: I've removed the dependencies

**Review by izzappel — APPROVED** [review]

**Review by yilkur — APPROVED** [review]

**Pull request merged**